### PR TITLE
feat: Add users education experience

### LINKED ISSUE #412: allow admin to add user Educational experience
An admin should be able to add educational experience to the user profile
a user can have none, one or more educational experiences
it should have four fields 

institution name, course name, start and end date

### PR DESCRIPTION
**Pull request by gabrielcosta42**
Adds ability for admins to include users education experience.

image
image
image
image

Closes #412

### CODE REVIEW

File: app/models/education_experience.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    less_than: Date.current,
+    allow_nil: true
+
+  validates_comparison_of :end_date,
+    message: lambda { |education_experience, other|
+      I18n.t(
+        "activerecord.errors.models.education_experience.attributes.end_date.less_than_start_date",
+        date: I18n.l(education_experience.start_date)
+      )
```
Comment (adbatista) [inline app/models/education_experience.rb:21]: active record allows you to customize the message without having to pass here

File: app/admin/education_experiences.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    column :institution
+    column :course
+    column :start_date
+    column :end_date
+
+    actions
+  end
+
+  form do |f|
+    user_id = params[:user_id] || f.object.user_id
```
Comment (adbatista) [inline app/admin/education_experiences.rb:21]: when is the `params[:user_id]` used?

Comment (gabrielcosta42) [reply app/admin/education_experiences.rb:21]: It's for adding new education experiences by the user experience tab.

File: app/models/education_experience.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  validates_presence_of :institution, :course, :start_date, :end_date
+
+  validates_comparison_of :start_date,
+    message: lambda { |education_experience, other|
+      I18n.t(
+        "activerecord.errors.models.education_experience.attributes.start_date.greater_than_current",
+        date: I18n.l(Date.current)
+      )
+    },
+    less_than: Date.current,
```
[the remark this answers is not in the document]
Comment (gabrielcosta42) [reply app/models/education_experience.rb]: done

[the remark this answers is not in the document]
Comment (gabrielcosta42) [reply app/models/education_experience.rb]: Done

File: app/models/education_experience.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+class EducationExperience < ApplicationRecord
+  belongs_to :user
+
+  validates_presence_of :institution, :course, :start_date, :end_date
+
+  validates_comparison_of :start_date,
+    message: lambda { |education_experience, other|
```
[the remark this answers is not in the document]
Comment (gabrielcosta42) [reply app/models/education_experience.rb]: the lambda wont be necessary anymore

File: app/models/education_experience.rb (hunk 1 above)
Comment (gabrielcosta42) [reply app/models/education_experience.rb]: done

**Review by adbatista — APPROVED** [review]

**Pull request merged**